lib/modules/dosomething/dosomething_reportback/dosomething_reportback_item.query.inc

```diff
@@ -117,23 +117,35 @@ function dosomething_reportback_build_reportback_items_query($params = array())
 *  - offset (int)
 *  - random (bool)
 *  - load_user (bool)
+* @param  bool  $tally
 * @return object
 *    An executed database query object to iterate through.
 */
-function dosomething_reportback_get_reportback_items_query($params) {
+function dosomething_reportback_get_reportback_items_query($params, $tally = FALSE) {
   if (isset($params['status']) && !$params['status']) {
       return FALSE;
   }
 
   $query = dosomething_reportback_build_reportback_items_query($params);
   $offset = dosomething_helpers_isset($params, 'offset', 0);
-  $count = dosomething_helpers_isset($params, 'count', 25);
+
+  if ($tally) {
+    $count = 'all';
+  }
+  else {
+    $count = dosomething_helpers_isset($params, 'count', 25);
+  }
 
   if ($count && $count !== 'all') {
     $query->range($offset, $count);
```

Comment:
my brain is bleeding. 
but, I _think_ you could move this `$query->range($offset, $count);` into the above else to avoid checking `if ($tally)` twice
and then the logic would read

```
if ($tally) 
  $count = $all;
  $query->execute;
  return $result ->rowCount()

else {
  the rest with the fetch all
```